modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -77,6 +166,47 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
     Events.fireTableModified(editor, table.dom, data.effect);
   };
 
+  const toggleTableClass = (_ui: boolean, requestedClass: string) => {
+    getSelectionStartCellOrCaption(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        if (Class.has(table, requestedClass)) {
+          Class.remove(table, requestedClass);
+        } else {
+          Class.add(table, requestedClass);
+        }
+
+        Events.fireTableModified(editor, table.dom, Events.structureModified);
+      });
+    });
+  };
+
+  const toggleTableCellClass = (_ui: boolean, requestedClass: string) => {
+    getSelectionStartCell(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        const cells = TableSelection.getCellsFromSelection(startCell, selections);
+
+        const warehouse = Warehouse.fromTable(table);
```

Comment:
`Warehouse` is not meant to be used outside snooker for things that modify the table, it's only exposed to query things. Anything that's modifying it really should be moved to snooker, so could you please look at doing that for this?

Reply:
Actually looking more, this doesn't even need to use the `Warehouse`. It could just use `TableLookup.cells` instead, so please use that instead here.